server/src/main/java/io/crate/planner/optimizer/symbol/rule/SwapCastsInComparisonOperators.java

```diff
@@ -71,7 +71,7 @@ public Symbol apply(Function operator,
         var literalOrParam = operator.arguments().get(1);
         var castFunction = captures.get(castCapture);
         var reference = castFunction.arguments().get(0);
-        DataType<?> targetType = reference.valueType();
+        DataType<?> targetType = castFunction.valueType();
```

Comment:
This caused `numeric(precision, scale)` to be moved to the literal instead of `numeric` cast.

Reply:
This change uncovers another bug,
```
cr> create table t (a float);
CREATE OK, 1 row affected (1.738 sec)
cr> insert into t values (0.123456789);
INSERT OK, 1 row affected (0.155 sec)
cr> select * from t where a = 0.123456789;
+------------+
|          a |
+------------+
| 0.12345679 |  -- this should not be returned
+------------+
SELECT 1 row in set (0.033 sec)
```
What happens is that `a = 0.123456789` > `a::double = 0.123456789` > `a = 0.123456789::float`(_wrong cast moved_) > `a = 0.12345679` and falsely matches the row.